### PR TITLE
containerization: Deprecate Ip from tablet record

### PR DESCRIPTION
**Pull request by sougou**
The IP does not need to be stored in the tablet record. Most code
should use the hostname instead. If the IP is really needed, it's
better to resolve it from the hostname.

This is prep work for introducing MySQLHostname which will not
store an IP either. This way, it will be less confusing if no
IP is cached.

### CODE REVIEW

File: proto/topodata.proto [hunk 1]
```diff
@@ -110,7 +110,7 @@ message Tablet {
   // Fully qualified domain name of the host.
   string hostname = 2;
 
-  // IP address, stored as a string.
+  // ip is deprecated. It is instead computed as needed from hostname.
```
Comment (alainjobart) [inline proto/topodata.proto]: Should comment out the field, and add:
reserved 3;

Comment (sougou) [inline proto/topodata.proto]: Done.

**Comment by alainjobart** [comment]
LGTM but not sure about the travis failures

**Comment by sougou** [comment]
Two of them look like a travis problem. So, I'm running the full suite on my box as we speak, just to make sure.

**Comment by sougou** [comment]
Tests are kind of passing on my box. I see some flakyness that doesn't look related. So, I'm going to optimistically merge this.

File: go/vt/topotools/tablet.go [hunk 2]
```diff
@@ -67,36 +67,19 @@ func ChangeType(ctx context.Context, ts topo.Server, tabletAlias *topodatapb.Tab
...
-// CheckOwnership returns nil iff the IP and port match on oldTablet and
+// CheckOwnership returns nil iff the Hostname and port match on oldTablet and
 // newTablet, which implies that no other tablet process has taken over the
 // record.
 func CheckOwnership(oldTablet, newTablet *topodatapb.Tablet) error {
 	if oldTablet == nil || newTablet == nil {
 		return errors.New("unable to verify ownership of tablet record")
 	}
-	if oldTablet.Ip != newTablet.Ip || oldTablet.PortMap["vt"] != newTablet.PortMap["vt"] {
+	fmt.Printf("Comparing:\n%v\n%v\n", oldTablet, newTablet)
```
Comment (michael-berlin) [inline go/vt/topotools/tablet.go]: Sugu, this was probably meant for debugging? Can you please remove it?

Comment (sougou) [inline go/vt/topotools/tablet.go]: Yeah. I fixed it in the next submit: https://github.com/youtube/vitess/commit/0fa6aa35ebc47ca9b373473c1ce1f993bb0a8cf5

**Pull request merged**